api/src/main/java/org/apache/iceberg/util/CharSequenceSet.java

```diff
@@ -54,10 +54,21 @@ public boolean isEmpty() {
     return wrapperSet.isEmpty();
   }
 
+  public CharSequenceWrapper wrapper() {
```

Comment:
Maybe you can use guava `Suppliers.memoize` to help lazy-init.

Reply:
Thanks for pointing this out. I didn't know about `Suppliers.memoize` and it looks useful.

I'm not sure that we want to use it here. We've had quite a few problems with Guava and have finally decided to shade just the classes we use into an Iceberg module. I think it is probably better to avoid adding to the set of Guava classes that we use. Since this is already done and it doesn't look like there is much of a benefit to changing over to Guava, I'm inclined to leave it as it is.

If others feel strongly about this, I can use it, but I think we should avoid expanding our use of Guava.

Reply:
Sounds good to me

Reply:
Sounds good to me